wellcomeml/datasets/winer.py

```diff
@@ -0,0 +1,279 @@
+                        # this string into a list of numbers for each sentence
+                        sent_wordidx = [l.split(' ') for l in sentences]
+                        # Go through each of the entites for this article
+                        # and translate the word ids to their words to create
+                        # the tokens and each entity.
+                        for sentid, sentence_entities in article_entities.items():
+                            if len(sent_wordidx) > int(sentid):
+                                # Translate the list of words ids for this sentence into 
+                                # the words using id2word
+                                tokens = [id2word[int(s)] for s in sent_wordidx[int(sentid)]]
+                                # Every word in the sentence is classed as an outside word 
+                                # until found otherwise from the entity list
+                                tags = ['O']*len(tokens)
+                                for begin, end, ent_type in sentence_entities:
+                                    # For entities that spread over multiple words
+                                    # it should be clear which word is the start of the 
+                                    # entity and which is the end
+                                    tags[(begin+1):(end-1)] = [str(ent_type) + '-I']*(end-1-begin-1)
+                                    tags[end-1] = str(ent_type) + '-E'
+                                    tags[begin] = str(ent_type) + '-B'
+                                for token, tag in list(zip(tokens, tags)):
+                                    output_file.write(' '.join([token, tag]))
+                                    output_file.write('\n')
+                            output_file.write('\n')
+
+def _load_data_spacy(data_path, inc_outside=True, merge_entities=True):
+
+    # Load data in Spacy format:
+    # X = list of sentences (plural) / documents ['the cat ...', 'some dog...', ...]
+    # Y = list of list of entity tags for each sentence [[{'start': 36, 'end': 46, 'label': 'PERSON'}, {..}, ..], ... ]
+    # inc_outside = False: don't include none-entities in the output
+    # merge_entities if entities span over multiple tags do you want to merge them or not
+    # e.g. John 3-B Smith 3-E -> John Smith 3
+
+    X = []
+    Y = []
+    with open(data_path) as f:
+        lines = f.read().split('ID ')
+        for line in lines:
+            if line != '':
+                sentences = line.split('\n\n')
+                # The first and last sentence is blank due to splitting
+                sentences = sentences[1:-1]
+                for sentence in sentences:
+                    char_i = 0 # A counter to populate the start and end character indexes for each entity
+                    sentence_text = ''
+                    sentence_tags = []
+                    entities = sentence.split('\n') # ['Gibraltar 3-B', 'has O', 'a O', ...]
```

Comment:
it would possibly be easier to iterate over lines in the first place and just identify places whereas two successive newlines occured